src/discretization/fist.jl

```diff
@@ -50,8 +50,13 @@ function discretize(polyarea::PolyArea, ::FIST)
       𝒫 = Chain(points[inds])
       n = nvertices(𝒫)
       # 3. update 𝒬 near clipped ear
-      isear(𝒫, i)   && (𝒬 = 𝒬 ∪ [mod1(i,n)])
-      isear(𝒫, i+1) && (𝒬 = 𝒬 ∪ [mod1(i+1,n)])
+      for j in (i-1, i)
+        if isear(𝒫, j)
+          𝒬 = 𝒬 ∪ [mod1(j,n)]
+        else
+          𝒬 = setdiff(𝒬, [mod1(j,n)])
+        end
+      end
```

Comment:
Quick note: this is not in-place, so likely slower that the original PR with unnecessary allocations. But we can do an optimization pass later and not worry about that now.

Reply:
Yep. I think we can work on performance optimizations in a separate pass. We are still in the process of making sure the algorithm is working smoothly with various example polygons.